### PR TITLE
Added FB authentication via django-allauth library

### PR DESCRIPTION
**Pull request by csuksangium**
added facebook authentication to page, available @ localhost:8000/accounts/login through django allauth library
after successfully logging in, user data is stored in database and page is redirected to /restaurants

### CODE REVIEW

**Comment by csuksangium** [comment]
note that manage.py needs to be migrated before fb login works

**Comment by csuksangium** [comment]
http://django-allauth.readthedocs.io/en/latest/installation.html for full installation guide

File: cs411f17/settings.py [hunk 1]
```diff
@@ -38,7 +38,14 @@
     'django.contrib.messages',
     'django.contrib.staticfiles',
     'testapp.apps.TestappConfig',
-    'rest_framework'
+    'rest_framework',
+
```
Comment (freddiev4) [inline cs411f17/settings.py]: Extra newline here

File: cs411f17/settings.py [hunk 2]
```diff
@@ -64,11 +71,26 @@
...
 ]
 
+AUTHENTICATION_BACKENDS = (
+
+    # Needed to login by username in Django admin, regardless of `allauth`
+    'django.contrib.auth.backends.ModelBackend',
+
+    # `allauth` specific authentication methods, such as login by e-mail
+    'allauth.account.auth_backends.AuthenticationBackend',
+
```
Comment (freddiev4) [inline cs411f17/settings.py]: extra newline here and on line 81 & 84

**Review by freddiev4 — COMMENTED** [review]
Hey @csuksangium great work. Just had some small style comments here for you.

Can you also just add a screenshot or gif of what one should see when they use this?

**Comment by csuksangium** [comment]
removed the redundant empty lines and added .gif of log in flow

**Comment by csuksangium** [comment]
removed the gif file and squashed commits. link to .gif is here: https://imgur.com/a/wQIG7

**Review by freddiev4 — APPROVED** [review]

**Pull request merged**